_source/_posts/2021-09-20-cli-ssh-oauth-device-grant.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: blog_post
+title: "Command Line Login with the OAuth Device Grant"
+author: [huan-liu,  lars-johansen]
+by: internal-contributor
+communities: [mobile]
+description: "Tutorial: Learn how to log into input limited devices like TVs and terminals using the OAuth 2.0 Device Grant!"
+tags: [oauth, oauth2, security, mobile, oauth2-device-flow, device-flow, ssh, c]
+tweets:
+- "Here's how to do command line login for input-constrained devices, using the OAuth 2.0 Device Authorization Grant. Stream on!"
+- "Command line sign-in to input-constrained devices—check it out and tell us what you think!"
+#image: blog/cli-ssh-oauth-device-grant/social.jpg
+type: conversion
+---
+
+Even if you've never heard of the OAuth 2.0 Device Authorization Grant, formerly known as the Device Flow, you've probably seen it in action on a smart TV or other streaming device. The Device Authorization Grant is commonly used on devices with limited keyboard input ability. The process allows you to log in to an account more easily via a separate browser on your laptop, mobile phone, or other companion rich-input device.
+
+For instance, if AtkoTV is launching a streaming service; its streaming Apple TV app might present you with something like the sign-in screen below. Instead of struggling to input anything on your TV, you'd log in on your laptop or mobile phone to complete the sign-in process.
```

Comment:
```suggestion
For instance, if AtkoTV is launching a streaming service, its streaming Apple TV app might present you with something like the sign-in screen below. Instead of struggling to input anything on your TV, you'd log in on your laptop or mobile phone to complete the sign-in process.
```